### PR TITLE
Add simpleNames option

### PR DESCRIPTION
**Pull request by Alan-Cha**

### CODE REVIEW

File: packages/openapi-to-graphql/src/oas_3_tools.ts [hunk 1]
```diff
@@ -472,22 +484,23 @@ export function getSchemaTargetGraphQLType(
...
+  const pathNoPathParams = path.split('/').reduce((path, part) => {
     if (!/{|}/g.test(part)) {
       return path + capitalize(part)
     } else {
       return path
     }
   })
 
-  return pathNoPathParams
+  return uncapitalize(pathNoPathParams)
```
Comment (Alan-Cha) [inline packages/openapi-to-graphql/src/oas_3_tools.ts]: I was getting some strange behavior from this which caused problem in other places in the code. I am enforcing the lowercase letter for now but I'm not sure if this is the right way about going at it.

File: packages/openapi-to-graphql/src/resolver_builder.ts [hunk 2]
```diff
@@ -143,13 +145,16 @@ export function getResolver({
...
+        paramName,
+        !data.options.simpleNames
+          ? Oas3Tools.CaseStyle.camelCase
+          : Oas3Tools.CaseStyle.simple
+      )
+
       let value = argsFromLink[paramName]
 
-      let paramNameWithoutLocation = paramName
-      if (paramName.indexOf('.') !== -1) {
```
Comment (Alan-Cha) [inline packages/openapi-to-graphql/src/resolver_builder.ts]: This didn't make any sense to me. Please let me know if you can make any sense from it. I have deleted it for now.

Comment (Alan-Cha) [reply packages/openapi-to-graphql/src/resolver_builder.ts]: I think this is in reference to the parameter names in link objects. This is what a typical link object looks like:

```
      "Author": {
        "$ref": "ThisIsHereAndFormattedThisWayBecauseOfAnIssueWithADependency",
        "operationRef": "Example API 3#/paths/~1authors~1{authorId}/get",
        "parameters": {
          "authorId": "$request.path.username"
        },
        "description": "Link between two different APIs"
      }
```

I don't see why we need to check for `.` and possibly split it.

File: packages/openapi-to-graphql/src/oas_3_tools.ts [hunk 3]
```diff
@@ -280,43 +280,50 @@ function buildUrl(server: ServerObject): string {
...
 
 /**
- * Returns object | array where all object keys are sanitized. Keys passed in
- * exceptions are not sanitized.
+ * Returns object/array/scalar where all object keys (if applicable) are
+ * sanitized.
  */
-export function sanitizeObjKeys(
-  obj: object | Array<any>,
-  exceptions: string[] = []
```
Comment (Alan-Cha) [inline packages/openapi-to-graphql/src/oas_3_tools.ts]: This `exceptions` parameter is never used except for in one unit test. I don't see how we will ever support it so I will remove it for now.

File: packages/openapi-to-graphql/test/oas_3_tools.test.ts [hunk 4]
```diff
@@ -74,20 +74,6 @@ test('Sanitize object keys when given an array', () => {
   ])
 })
 
-test('Sanitize object keys, but not $ref', () => {
```
Comment (Alan-Cha) [inline packages/openapi-to-graphql/test/oas_3_tools.test.ts]: This is the one unit test I was referring to.

File: packages/openapi-to-graphql/src/oas_3_tools.ts [hunk 5]
```diff
@@ -280,43 +280,50 @@ function buildUrl(server: ServerObject): string {
...
+
   return cleanKeys(obj)
 }
 
 /**
  * Desanitizes keys in given object by replacing them with the keys stored in
  * the given mapping.
  */
-export function desanitizeObjKeys(
+export function desanitizeObjectKeys(
```
Comment (Alan-Cha) [inline packages/openapi-to-graphql/src/oas_3_tools.ts]: We actually have another function called `desanitizeObjectKeys()` in the same file...

I realized that the other `desanitizeObjectKeys()` function was not useful so I ended up removing it but I was surprised to find it.

File: packages/openapi-to-graphql/src/oas_3_tools.ts [hunk 6]
```diff
@@ -280,43 +280,50 @@ function buildUrl(server: ServerObject): string {
...
- * exceptions are not sanitized.
+ * Returns object/array/scalar where all object keys (if applicable) are
+ * sanitized.
  */
-export function sanitizeObjKeys(
-  obj: object | Array<any>,
-  exceptions: string[] = []
-): object | Array<any> {
-  const cleanKeys = (obj: object | Array<any>): object | Array<any> => {
+export function sanitizeObjectKeys(
```
Comment (Alan-Cha) [inline packages/openapi-to-graphql/src/oas_3_tools.ts]: We actually have another function called `sanitizeObjectKeys()` in the same file...

I realized that the other `sanitizeObjectKeys()` function was not useful so I ended up removing it but I was surprised to find it.

File: packages/openapi-to-graphql/src/oas_3_tools.ts [hunk 7]
```diff
@@ -1096,17 +1118,6 @@ export function storeSaneName(
   return saneStr
 }
 
-/**
- * Return an object similar to the input object except the keys are all
- * sanitized
- */
-export function sanitizeObjectKeys(obj: object): object {
```
Comment (Alan-Cha) [inline packages/openapi-to-graphql/src/oas_3_tools.ts]: The useless `sanitizeObjectKeys()` function I mentioned.

File: packages/openapi-to-graphql/src/oas_3_tools.ts [hunk 8]
```diff
@@ -280,43 +280,50 @@ function buildUrl(server: ServerObject): string {
...
-          if (Object.prototype.hasOwnProperty.call(obj, key)) {
-            res[saneKey] = cleanKeys(obj[key])
-          }
-        } else {
-          res[key] = cleanKeys(obj[key])
+
+      for (const key in obj) {
+        const saneKey = sanitize(key, caseStyle)
+
+        if (Object.prototype.hasOwnProperty.call(obj, key)) {
```
Comment (Alan-Cha) [inline packages/openapi-to-graphql/src/oas_3_tools.ts]: Is this necessary?

File: packages/openapi-to-graphql/lib/types/options.d.ts [hunk 9]
```diff
@@ -62,7 +62,7 @@ export declare type Options = {
      * Allows to define the root operation type (Query or Mutation type) of any
      * OAS operation explicitly.
      *
-     * OtG will by default make all GET operations Query fields and all other
+     * OTG will by default make all GET operations Query fields and all other
```
Comment (Alan-Cha) [inline packages/openapi-to-graphql/lib/types/options.d.ts]: Consider between `OTG` and `OtG`.

File: packages/openapi-to-graphql/src/oas_3_tools.ts (hunk 1 above)
Comment (Alan-Cha) [reply packages/openapi-to-graphql/src/oas_3_tools.ts]: Remove this.

File: packages/openapi-to-graphql/src/preprocessor.ts [hunk 10]
```diff
@@ -532,7 +538,14 @@ export function createDataDef(
       const name = getSchemaName(names, data.usedTypeNames)
 
       // Store and sanitize the name
-      const saneName = Oas3Tools.sanitize(name, Oas3Tools.CaseStyle.PascalCase)
+      let saneName
```
Comment (Alan-Cha) [inline packages/openapi-to-graphql/src/preprocessor.ts]: Use this pattern:

```
            !data.options.simpleNames
              ? Oas3Tools.CaseStyle.camelCase
              : Oas3Tools.CaseStyle.simple
```

**Comment by ErikWittern** [comment]
@Alan-Cha Thanks for addressing the comments. I approve of this PR.

**Pull request merged**